### PR TITLE
cpu/esp_common: esp_now doesn't call memcpy if iol_len is 0

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

Fixes the problem described in #11163.

### Testing procedure

Tested with the procedure described in #11163 

### Issues/PRs references

Related to issue #11163

### CODE REVIEW

File: cpu/esp_common/esp-now/esp_now_netdev.c [hunk 1]
```diff
@@ -652,11 +652,11 @@ static int _send(netdev_t *netdev, const iolist_t *iolist)
                   data_len + iolist->iol_len, ESP_NOW_MAX_SIZE_RAW);
```
Comment (miri64) [inline cpu/esp_common/esp-now/esp_now_netdev.c]: Can it be guaranteed that [this dereference above](https://github.com/RIOT-OS/RIOT/pull/11187/files#diff-24495fdc92b602d8c37d024123b775a5R636) does not dereference a NULL pointer?

Comment (gschorcht) [reply cpu/esp_common/esp-now/esp_now_netdev.c]: It is guaranteed because this part of the `iol_list` is created and filled by https://github.com/RIOT-OS/RIOT/blob/71204d1d7034b642d310604d0bc9ec4fe634f414/cpu/esp_common/esp-now/esp_now_gnrc.c#L69-L73 as variable on stack before it calls esp_now_netdev::_send with the `iol_list`.

Comment (miri64) [reply cpu/esp_common/esp-now/esp_now_netdev.c]: But this is only true for GNRC. If another (hypothetical) stack does not necessarily provide this, this can lead to problems. Better document this in the [`esp-now` documentation](https://doc.riot-os.org/esp__now__netdev_8h.html) somewhere.

Comment (miri64) [reply cpu/esp_common/esp-now/esp_now_netdev.c]: (not a blocker for this PR, as it should go in a separate PR)

Comment (miri64) [reply cpu/esp_common/esp-now/esp_now_netdev.c]: Though... I just noticed: Having at least a non-zero L2 header should be a requirement for most platforms. Maybe we rather do it the other way around: document that the device *may* need the first header to be of non-zero length, and document the corner cases, where this isn't the case necessarily (e.g. SLIP).

**Comment by miri64** [comment]
LGTM, but can not test.

**Comment by gschorcht** [comment]
@miri64 Added the proposed NULL pointer check to this PR. I hope it's ok, otherwise I would have to prepare a separate PR for a single line.

File: cpu/esp_common/esp-now/esp_now_netdev.c [hunk 2]
```diff
@@ -633,7 +633,7 @@ static int _send(netdev_t *netdev, const iolist_t *iolist)
     uint8_t* _esp_now_dst = NULL;
 
     for (uint8_t i = 0; i < ESP_NOW_ADDR_LEN; i++) {
-        if (((uint8_t*)iolist->iol_base)[i] != 0xff) {
+        if (iolist->iol_base != NULL && ((uint8_t*)iolist->iol_base)[i] != 0xff) {
```
Comment (miri64) [inline cpu/esp_common/esp-now/esp_now_netdev.c]: parenthesis missing around expressions.

**Comment by miri64** [comment]
Looks even better. Still don't have the hardware to test :(

File: cpu/esp_common/esp-now/esp_now_netdev.c (hunk 2 above)
Comment (gschorcht) [reply cpu/esp_common/esp-now/esp_now_netdev.c]: Ok, even though the compiler seems to be happy and doesn't warn.

Comment (gschorcht) [reply cpu/esp_common/esp-now/esp_now_netdev.c]: Changed it.

Comment (miri64) [reply cpu/esp_common/esp-now/esp_now_netdev.c]: > Ok, even though the compiler seems to be happy and doesn't warn.

It's not about the compiler but about [readability / style](https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions#indentation-and-braces) ;-).

File: cpu/esp_common/esp-now/esp_now_netdev.c [hunk 3]
```diff
@@ -633,7 +633,7 @@ static int _send(netdev_t *netdev, const iolist_t *iolist)
     uint8_t* _esp_now_dst = NULL;
 
     for (uint8_t i = 0; i < ESP_NOW_ADDR_LEN; i++) {
-        if (((uint8_t*)iolist->iol_base)[i] != 0xff) {
+        if ((iolist->iol_base != NULL) && ((uint8_t*)iolist->iol_base)[i] != 0xff) {
```
Comment (miri64) [inline cpu/esp_common/esp-now/esp_now_netdev.c]: Still parentheses missing around the second expression.

Comment (gschorcht) [reply cpu/esp_common/esp-now/esp_now_netdev.c]: Ups, I should have taken more time to fix it :sunglasses:

**Comment by miri64** [comment]
I'm fine with the fixups. You may squash. I'll try to get some hardware from my colleagues next week / try to convince them to test.

**Comment by gschorcht** [comment]
@miri Squashed. Thanks for reviewing.

**Comment by gschorcht** [comment]
@miri64 Thanks for reviewing it.
